include/unifex/tag_invoke.hpp

```diff
@@ -42,26 +66,23 @@ using namespace tag_invoke_cpo_ns;
 template <auto& CPO>
 using tag_t = std::remove_cvref_t<decltype(CPO)>;
 
-template <typename CPO, typename... Args>
-using tag_invoke_result = std::invoke_result<tag_t<tag_invoke>, CPO, Args...>;
+using tag_invoke_impl::tag_invoke_result_t;
 
-template <typename CPO, typename... Args>
-using tag_invoke_result_t =
-    std::invoke_result_t<tag_t<tag_invoke>, CPO, Args...>;
+template<typename CPO, typename... Args>
+struct tag_invoke_result {
+    using type = tag_invoke_result_t<CPO, Args...>;
+};
 
 template <typename CPO, typename... Args>
-inline constexpr bool is_tag_invocable_v =
-    std::is_invocable_v<tag_t<tag_invoke>, CPO, Args...>;
+using is_tag_invocable = typename tag_invoke_impl::is_tag_invocable<CPO, Args...>::type;
 
 template <typename CPO, typename... Args>
-using is_tag_invocable = std::is_invocable<tag_t<tag_invoke>, CPO, Args...>;
+inline constexpr bool is_tag_invocable_v = is_tag_invocable<CPO, Args...>::value;
```

Comment:
Instantiating variable templates is *much* cheaper than instantiating class templates. The class templates should be defined in terms of the variable templates, not the other way 'round.

Reply:
The implementation of this has already produced the result as a `std::bool_constant`.
I would need to change the implementation of the `try_call()` mechanism in `tag_invoke_impl::is_tag_invocable` to avoid this.

Any suggestions? Maybe using `sizeof(decltype(try_call(0)) == sizeof(yes_type)`?

Reply:
The `sizeof()` approach seems to work.

I moved the `try_call` function overloads to namespace scope (and renamed to `try_tag_invoke`).
I also combined the `noexcept` test into the same function instantiation.